packages/flutter/lib/src/gestures/pointer_signal_resolver.dart

```diff
@@ -47,7 +51,7 @@ class PointerSignalResolver {
       assert(_currentEvent == null);
       return;
     }
-    assert((_currentEvent.original ?? _currentEvent) == event);
+    assert(_isSameEvent(_currentEvent, event));
```

Comment:
@goderbauer @stuartmorgan is `resolve()` supposed to be called only with the original event?

Reply:
This assertion is here because pointer signals are resolved synchronously, and thus there was no expectation that there would be more than one event being processed at any one time. After discussion with Hixie, the decision was to use a single event rather than a set, and assert that nothing was violating that invariant.

> If two hit targets receive a PointerSignalEvent, they both try to call `PointerSignalResolver.register()` but they pass different events (each target receives its own transformed event).

I'm not familiar with transformed events, is it just local coordinates per target or something along those lines? I.e., are they still a single conceptual event, which only one item should be claiming? If so, then this change should be fine. If there are multiple conceptual events at once then a more substantial change is required.

Reply:
> is it just local coordinates per target or something along those lines?

That's exactly right.

Michael made a change few months ago so each hit target receives a "transformed" event. The transformer event keeps track of the original event it was created from. What I'm doing here is checking if the events have the `original` event (I.e. they were created from the same event).

Reply:
IOW, yes the two events are generated from the same conceptual event.

Reply:
> yes the two events are generated from the same conceptual event.

This is correct. And for the intents of the SignalRessolver they should be treated as the same event.

They way "resolve" is currently called grantees that it will always get the untransformed original event. However, I think it's fine to loosen that constraint as is done in this PR.